### ISSUE TITLE
Issue with gifski_renderer: flashing green background (magick_renderer works)

### ISSUE THREAD

**Issue by LukasWallrich** [thomasp85/gganimate#431]
Thanks for this wonderful package. 

On my system, I cannot render animations with the `gifski_renderer` correctly (they flash green), while the `magick_renderer` works fine. See below for two reprexes with their respective results. If this could be fixed, that would be wonderful since the magick_renderer does not support `ease_aes` - any suggestions?

``` r
library(magrittr)
library(ggplot2)
library(gganimate)
#> Warning: package 'gganimate' was built under R version 4.0.4
anim <- mtcars %>% ggplot(aes(x = wt, y = mpg)) +
  geom_point() +
  transition_states(cyl, transition_length = 1, state_length = 2)

animate(anim, renderer = gifski_renderer())
```

![](https://i.imgur.com/OhuHFOs.gif)

<sup>Created on 2021-03-03 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
<details>
<summary>
Session info
</summary>

``` r
devtools::session_info()
#> - Session info ---------------------------------------------------------------
#>  setting  value                       
#>  version  R version 4.0.3 (2020-10-10)
#>  os       Windows 10 x64              
#>  system   x86_64, mingw32             
#>  ui       RTerm                       
#>  language (EN)                        
#>  collate  English_United Kingdom.1252 
#>  ctype    English_United Kingdom.1252 
#>  tz       Europe/London               
#>  date     2021-03-03                  
#> 
#> - Packages -------------------------------------------------------------------
#>  package     * version date       lib source        
#>  assertthat    0.2.1   2019-03-21 [1] CRAN (R 4.0.0)
#>  callr         3.5.1   2020-10-13 [1] CRAN (R 4.0.3)
#>  cli           2.2.0   2020-11-20 [1] CRAN (R 4.0.3)
#>  colorspace    2.0-0   2020-11-11 [1] CRAN (R 4.0.3)
#>  crayon        1.3.4   2017-09-16 [1] CRAN (R 4.0.0)
#>  curl          4.3     2019-12-02 [1] CRAN (R 4.0.0)
#>  desc          1.2.0   2018-05-01 [1] CRAN (R 4.0.0)
#>  devtools      2.3.2   2020-09-18 [1] CRAN (R 4.0.3)
#>  digest        0.6.27  2020-10-24 [1] CRAN (R 4.0.3)
#>  dplyr         1.0.2   2020-08-18 [1] CRAN (R 4.0.3)
#>  ellipsis      0.3.1   2020-05-15 [1] CRAN (R 4.0.0)
#>  evaluate      0.14    2019-05-28 [1] CRAN (R 4.0.0)
#>  fansi         0.4.1   2020-01-08 [1] CRAN (R 4.0.0)
#>  farver        2.0.3   2020-01-16 [1] CRAN (R 4.0.0)
#>  fs            1.5.0   2020-07-31 [1] CRAN (R 4.0.2)
#>  generics      0.1.0   2020-10-31 [1] CRAN (R 4.0.3)
#>  gganimate   * 1.0.7   2020-10-15 [1] CRAN (R 4.0.4)
#>  ggplot2     * 3.3.2   2020-06-19 [1] CRAN (R 4.0.1)
#>  gifski        0.8.7   2021-02-28 [1] CRAN (R 4.0.4)
#>  glue          1.4.2   2020-08-27 [1] CRAN (R 4.0.2)
#>  gtable        0.3.0   2019-03-25 [1] CRAN (R 4.0.0)
#>  highr         0.8     2019-03-20 [1] CRAN (R 4.0.0)
#>  hms           0.5.3   2020-01-08 [1] CRAN (R 4.0.0)
#>  htmltools     0.5.1.1 2021-01-22 [1] CRAN (R 4.0.3)
#>  httr          1.4.2   2020-07-20 [1] CRAN (R 4.0.2)
#>  knitr         1.30    2020-09-22 [1] CRAN (R 4.0.2)
#>  labeling      0.4.2   2020-10-20 [1] CRAN (R 4.0.3)
#>  lifecycle     0.2.0   2020-03-06 [1] CRAN (R 4.0.0)
#>  magrittr    * 2.0.1   2020-11-17 [1] CRAN (R 4.0.3)
#>  memoise       1.1.0   2017-04-21 [1] CRAN (R 4.0.0)
#>  mime          0.9     2020-02-04 [1] CRAN (R 4.0.0)
#>  munsell       0.5.0   2018-06-12 [1] CRAN (R 4.0.0)
#>  pillar        1.4.7   2020-11-20 [1] CRAN (R 4.0.3)
#>  pkgbuild      1.2.0   2020-12-15 [1] CRAN (R 4.0.3)
#>  pkgconfig     2.0.3   2019-09-22 [1] CRAN (R 4.0.0)
#>  pkgload       1.1.0   2020-05-29 [1] CRAN (R 4.0.0)
#>  plyr          1.8.6   2020-03-03 [1] CRAN (R 4.0.0)
#>  prettyunits   1.1.1   2020-01-24 [1] CRAN (R 4.0.0)
#>  processx      3.4.5   2020-11-30 [1] CRAN (R 4.0.3)
#>  progress      1.2.2   2019-05-16 [1] CRAN (R 4.0.0)
#>  ps            1.5.0   2020-12-05 [1] CRAN (R 4.0.3)
#>  purrr         0.3.4   2020-04-17 [1] CRAN (R 4.0.0)
#>  R6            2.5.0   2020-10-28 [1] CRAN (R 4.0.3)
#>  Rcpp          1.0.6   2021-01-15 [1] CRAN (R 4.0.3)
#>  remotes       2.2.0   2020-07-21 [1] CRAN (R 4.0.2)
#>  rlang         0.4.10  2020-12-30 [1] CRAN (R 4.0.3)
#>  rmarkdown     2.6     2020-12-14 [1] CRAN (R 4.0.3)
#>  rprojroot     2.0.2   2020-11-15 [1] CRAN (R 4.0.3)
#>  scales        1.1.1   2020-05-11 [1] CRAN (R 4.0.0)
#>  sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 4.0.0)
#>  stringi       1.5.3   2020-09-09 [1] CRAN (R 4.0.2)
#>  stringr       1.4.0   2019-02-10 [1] CRAN (R 4.0.0)
#>  testthat      3.0.1   2020-12-17 [1] CRAN (R 4.0.3)
#>  tibble        3.0.4   2020-10-12 [1] CRAN (R 4.0.3)
#>  tidyselect    1.1.0   2020-05-11 [1] CRAN (R 4.0.0)
#>  tweenr        1.0.1   2018-12-14 [1] CRAN (R 4.0.0)
#>  usethis       2.0.0   2020-12-10 [1] CRAN (R 4.0.3)
#>  vctrs         0.3.6   2020-12-17 [1] CRAN (R 4.0.3)
#>  withr         2.4.0   2021-01-16 [1] CRAN (R 4.0.3)
#>  xfun          0.20    2021-01-06 [1] CRAN (R 4.0.3)
#>  xml2          1.3.2   2020-04-23 [1] CRAN (R 4.0.0)
#>  yaml          2.2.1   2020-02-01 [1] CRAN (R 4.0.0)
#> 
#> [1] C:/Users/lukas/Documents/R/win-library/4.0
#> [2] C:/Program Files/R/R-4.0.3/library
```

</details>

``` r
library(magrittr)
library(ggplot2)
library(gganimate)
#> Warning: package 'gganimate' was built under R version 4.0.4
anim <- mtcars %>% ggplot(aes(x = wt, y = mpg)) +
  geom_point() +
  transition_states(cyl, transition_length = 1, state_length = 2)

animate(anim, renderer = magick_renderer())
```

![](https://i.imgur.com/lbgKiDy.gif)

<sup>Created on 2021-03-03 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by jhelvy** [comment]
I'm getting the same issue. Just tested on a few different machines and all of them got the green glitches when using `gifski_renderer()` and they all worked fine with `magick_renderer()`. What was more confusing was that it appears to have used `gifski_renderer()` by default even if both `gifski` and `magick` are installed, so it wasn't clear what was causing the source of the problem.

**Comment by vidonne** [comment]
Same issue here with `gifski_renderer()` but the `magick_renderer()` didn't solve completely the issue for me. Some of the animations involving `geom_tile` still get the green background glitches. Any ideas on how to fix it?

**Comment by jeffreyblanchard** [comment]
I am having the same issue. Works with magick but not gifski.  The animation worked fine (gganimate), but then I installed gifski and the green glitches appeared.

**Comment by jeffreyblanchard** [comment]
The same issue with gifski occurs using RStudio Cloud

**Comment by mcanouil** [comment]
Mostly curious, what is the version of `gifski` you are using? 
I see in first post it is `v0.8.7`, but same for the others?
Apparently `v0.8.7`  was released ~15 days ago on CRAN (2021-02-28).

And the changelog for this version is not really reassuring:
```
0.8.7
  - CRAN insists that we override the standard cargo/rust behaviour which
    stores registry and crate data in ~/.cargo, because policy. So we now
    build with CARGO_HOME set to a temporary dir that is deleted afterwards.
    This probably makes this slower and more error prone, but I don't make
    these rules.
```

Maybe better to (cross-)post the issue in [r-rust/gifski](https://github.com/r-rust/gifski) (if it is only the gifski renderer which has the issue).

**Comment by vidonne** [comment]
@mcanouil I'm using `v0.8.7` too

**Comment by thomasp85** [comment]
@LukasWallrich can you open an issue in gifski? This is not a gganimate issue and there is nothing to do about it here unfortunately...

**Comment by jeffreyblanchard** [comment]
‘0.8.7.1’.  I just updated yesterday. Prior to that I had no green glitches in the gif when I ran this code.

**Comment by ravikrocha** [comment]
I got this message:
Error in if (loop) 0 else 1 : argument is not interpretable as logical

**Comment by LukasWallrich** [comment]
> @LukasWallrich can you open an issue in gifski? This is not a gganimate issue and there is nothing to do about it here unfortunately...

I have now opened a related issue in gifski - given that it was not yet reported there, it might be somehow related to gganimate though - let's see what they say: https://github.com/ImageOptim/gifski/issues/169

**Comment by thomasp85** [comment]
since it is version dependent on gifski it is most likely a gifski - if you want to be sure, you can use the file_renderer and inspect the generated png files - if they looks like they should then gganimate is doing everything it should

**Comment by LukasWallrich** [comment]
The issue indeed only arises with the updated gifski version. For those who are also affected, a workaround is to downgrade to a previous gifski version, e.g., by running `devtools::install_github("r-rust/gifski@ad925fe")`

**Comment by jeroen** [comment]
I can confirm it is a bug in the gifski 0.8.7. As a temporary workaround you can downgrade to gifski 0.8.6 like so:

```r
install.packages('gifski', repos = 'https://cran.microsoft.com/snapshot/2021-02-28')
```

**Comment by jeroen** [comment]
This problem is fixed in the new version gifski 1.4.3 which is now on CRAN.
